alot/crypto.py

```diff
@@ -154,18 +124,14 @@ def detached_signature_for(plaintext_str, key=None):
 
     :param plaintext_str: text to sign
     :param key: gpgme_key_t object representing the key to use
-    :rtype: tuple of gpgme.NewSignature array and str
+    :rtype: tuple of gpg.results.NewSignature array and str
     """
-    ctx = gpgme.Context()
+    ctx = gpg.Context()
     ctx.armor = True
     if key is not None:
         ctx.signers = [key]
-    plaintext_data = StringIO(plaintext_str)
-    signature_data = StringIO()
-    sigs = ctx.sign(plaintext_data, signature_data, gpgme.SIG_MODE_DETACH)
-    signature_data.seek(0, os.SEEK_SET)
-    signature = signature_data.read()
-    return sigs, signature
+    (sigblob, sign_result) = ctx.sign(gpg.Data(string=plaintext_str), mode=gpg.constants.SIG_MODE_DETACH)
```

Comment:
Actually, we wrap flos with a fileno() method and everything supporting the buffer protocol magically (in Python2 this includes strings, in Python3 bytestrings), no need for an explicit wrapping here.

... = ctx.sign(plaintext_str, ...) should suffice here.